frontend/src/app/models/user.ts

```diff
@@ -15,6 +20,7 @@ export class User {
         public name,
         public isAdmin: boolean,
         public downloadLimit: number = 0, // The download limit for this user, will be 0 if there is no limit.
-        public isSamlLogin: boolean
+        public isSamlLogin: boolean,
+        public enableSearchHistory: boolean,
```

Comment:
I'm unhappy with the distinction of the `UserResponse` and the `User` class in the frontend. I think I get the reasoning behind it: let the backend make the distinction, it does not matter for the frontend. However, there is a clear separation of responsibility, which I think should be reflected in the frontend. Also because it will only get more complex down the line (if there's more stuff in the profile) to make do the merging/splitting when communicating with the backend.

Reply:
I'm not sure I understand this. The distinction between `UserResponse` and `User` is between the API and the frontend model. I do think it's good practice to separate those representations. But I think this is about the user profile?